astropy/coordinates/sky_coordinate.py

```diff
@@ -663,6 +663,13 @@ def transform_to(self, frame, merge_attributes=True):
                      set(frame_kwargs.keys())):
             frame_kwargs.pop(attr)
 
+        # Always remove the origin frame attribute, as that attribute only makes
+        # sense with a SkyOffsetFrame (in which case it will be stored on the frame).
+        # See gh-11277.
+        # TODO: Should it be a property of the frame attribute that it can
+        # or cannot be stored on a SkyCoord?
+        frame_kwargs.pop('origin', None)
```

Comment:
Does this means that  ``SkyCoord(SkyOffsetFrame(...)).origin`` no longer works?

Reply:
https://github.com/astropy/astropy/pull/11730#issuecomment-841814187 Answers this.

Followup: given that users can make their own frames with arbitrary frame attributes, perhaps this special-casing of "origin" should be further special-cased by first checking if the 'from'-frame is `SkyOffsetFrame`

```suggestion
        if self.frame.__class__ is SkyOffsetFrame:
            frame_kwargs.pop('origin', None)
```

Reply:
You are right. I never liked this somewhat arbitrary popping of a name - really, `SkyOffsetFrame` itself should somehow tell that its attribute should not be propagated.